### PR TITLE
Caution flag reason text is displayed as input from CSV, along with other fields Fixes #5125

### PR DESCRIPTION
**Pull request by mphprogrammer**
Also removed ability to filter caution flags. Bad ju-ju.

### CODE REVIEW

File: spec/models/converters/base_converter_spec.rb [hunk 1]
```diff
@@ -14,15 +14,16 @@
     end
   end
 
+  it 'does not convert strings containing more than forbidden words to nil' do
```
Comment (patrickvinograd) [inline spec/models/converters/base_converter_spec.rb]: The spec description looks like it missing a word..."containing more than ??? forbidden words"

File: spec/models/converters/base_converter_spec.rb [hunk 2]
```diff
@@ -14,15 +14,16 @@
...
+  end
+
   it 'converts nil to nil' do
     expect(subject.convert(nil)).to be_nil
   end
 
   it 'converts blanks to an empty string' do
     expect(subject.convert('      ')).to be_blank
   end
-
```
Comment (patrickvinograd) [inline spec/models/converters/base_converter_spec.rb]: Instead of removing this test case, why not flip its logic to be "it preserves input case" and expect input to match output. That way if behavior changes later it is caught by the spec.

File: app/models/institution_builder.rb [hunk 3]
```diff
@@ -383,7 +383,7 @@ def self.add_hcm(version_number)
     str = <<-SQL
       UPDATE institutions SET
         caution_flag = TRUE,
-        caution_flag_reason = concat_ws(',', caution_flag_reason, hcm_list.reasons)
+        caution_flag_reason = concat_ws(', ', caution_flag_reason, hcm_list.reasons)
```
Comment (patrickvinograd) [inline app/models/institution_builder.rb]: Would be nice to add a spec covering this concatenation to make sure it looks right.

**Review by patrickvinograd — CHANGES_REQUESTED** [review]
Just a few minor spec issues. See comments.

File: app/controllers/v0/institutions_controller.rb [hunk 4]
```diff
@@ -60,7 +60,6 @@ def search_results
       @query ||= normalized_query_params
```
Comment (saneshark) [inline app/controllers/v0/institutions_controller.rb]: @mphprogrammer Do the normalized query params list caution_flag as an available query param? I think we should also maybe remove it from that array constant, if applicable.

File: app/models/complaint.rb [hunk 5]
```diff
@@ -75,7 +75,7 @@ class Complaint < ActiveRecord::Base
     'school state' => { column: :state, converter: BaseConverter },
     'submitted' => { column: :submitted, converter: BaseConverter },
     'closed' => { column: :closed, converter: BaseConverter },
-    'closed reason' => { column: :closed_reason, converter: BaseConverter },
+    'closed reason' => { column: :closed_reason, converter: DowncaseConverter },
```
Comment (saneshark) [inline app/models/complaint.rb]: Will these changes affect the CSV export vs what we show on the Front-end?

File: app/models/converters/boolean_converter.rb [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
 
 class BooleanConverter < BaseConverter
-  TRUTHS = %w(true t yes ye y 1 on).freeze
-
   def self.convert(value)
     value = super(value)
-    value.blank? ? nil : TRUTHS.include?(value.to_s)
+
+    return nil if value.blank?
+    value.to_s.match(/\A(true|t|yes|ye|y|1|on)\z/i).present? ? true : false
```
Comment (saneshark) [inline app/models/converters/boolean_converter.rb]: This is pretty awesome, but it might be something we would want to raise warnings around in the CSV message ultimately, maybe we can put a TODO comment or something in there. 

My ultimate goal would be to have the CSV that gets uploaded to be consistent. If there are this many freaking nuances to something characterized as true, the method whereby they are input is likely wrought with errors and the admins need to take a closer look at precisely what it is they are uploading.

Obviously this is just a nitpick for now, something to think about long-term.

File: app/models/hcm.rb [hunk 7]
```diff
@@ -10,7 +10,7 @@ class Hcm < ActiveRecord::Base
     'country' => { column: :country, converter: BaseConverter },
     'institution type' => { column: :institution_type, converter: BaseConverter },
     'stop pay/monitor method' => { column: :hcm_type, converter: BaseConverter },
-    'method reason desc' => { column: :hcm_reason, converter: DisplayConverter }
+    'method reason desc' => { column: :hcm_reason, converter: BaseConverter }
```
Comment (saneshark) [inline app/models/hcm.rb]: @mphprogrammer can you elaborate on why this change is necessary?

File: app/models/settlement.rb [hunk 8]
```diff
@@ -7,7 +7,7 @@ class Settlement < ActiveRecord::Base
     'instnm' => { column: :institution, converter: InstitutionConverter },
     'school_system_code' => { column: :school_system_code, converter: NumberConverter },
     'school_system_name' => { column: :school_system_name, converter: BaseConverter },
-    'settlement_description' => { column: :settlement_description, converter: DisplayConverter },
+    'settlement_description' => { column: :settlement_description, converter: BaseConverter },
```
Comment (saneshark) [inline app/models/settlement.rb]: Curious what this change helps facilitate.

**Review by saneshark — CHANGES_REQUESTED** [review]

File: app/models/complaint.rb (hunk 5 above)
Comment (mphprogrammer) [inline app/models/complaint.rb]: No, it shouldn't. I think that the exported data should represent exactly what is in the database at the time of export, with the same formatting and casing.

File: app/models/converters/boolean_converter.rb (hunk 6 above)
Comment (mphprogrammer) [inline app/models/converters/boolean_converter.rb]: I agree. We need to evangelize the need for consistent norms with data types in these CSVs. It would speed up processing.

File: app/models/hcm.rb (hunk 7 above)
Comment (mphprogrammer) [inline app/models/hcm.rb]: Yes - Patrick's issue was with the case of letters in acroynyms being an issue with the Dept of Ed. The BaseConverter leaves the case intact, but does strip quote characters, `null`, `none` and `privacysuppressed`, as well leading and trailing blanks. We are now displaying caution flag reasons exactly as input.

File: app/models/settlement.rb (hunk 8 above)
Comment (mphprogrammer) [inline app/models/settlement.rb]: Patrick's issue was with the case of letters in acroynyms being an issue. The BaseConverter leaves the case intact, but does strip quote characters, null, none and privacysuppressed, as well leading and trailing blanks. We are now displaying caution flag reasons exactly as input.

File: spec/models/converters/base_converter_spec.rb (hunk 1 above)
Comment (mphprogrammer) [inline spec/models/converters/base_converter_spec.rb]: Got it.

File: spec/models/converters/base_converter_spec.rb (hunk 2 above)
Comment (mphprogrammer) [inline spec/models/converters/base_converter_spec.rb]: Got it.

File: app/controllers/v0/institutions_controller.rb (hunk 4 above)
Comment (mphprogrammer) [inline app/controllers/v0/institutions_controller.rb]: Missed that! Its removed now.

**Review by patrickvinograd — APPROVED** [review]

**Comment by patrickvinograd** [comment]
My requested spec changes are accounted for, will let Kam take another pass.

File: app/models/institution_builder.rb (hunk 3 above)
Comment (mphprogrammer) [inline app/models/institution_builder.rb]: There is a spec in place for this that adds an accreditation caution first, and then an HCM. It uses a regex to match, which is why I missed the ',' rather than ', '. However, I did add a concatenation test for Accreditation - although it affects the caution flag reason first, if moved to another position in the build, the caution flag reason will still be concatenated properly.

**Review by saneshark — APPROVED** [review]

**Comment by saneshark** [comment]
@mphprogrammer LGTM
